src/mongo_client.ts

```diff
@@ -437,12 +436,57 @@ export class MongoClient extends TypedEventEmitter<MongoClientEvents> {
       throw new MongoInvalidArgumentError('Method `connect` only accepts a callback');
     }
 
-    return maybePromise(callback, cb => {
-      connect(this, this[kOptions], err => {
-        if (err) return cb(err);
-        cb(undefined, this);
-      });
-    });
+    return maybeCallback(async () => {
+      // If a connection already been established, we can terminate early
+      if (this.topology && this.topology.isConnected()) {
+        return this;
+      }
+
+      const options = this[kOptions];
+
+      if (typeof options.srvHost === 'string') {
+        const hosts = await resolveSRVRecord(options);
+
+        for (const [index, host] of hosts.entries()) {
+          options.hosts[index] = host;
+        }
```

Comment:
```suggestion
        options.hosts = await resolveSRVRecord(options);
```
any reason we can't do it this way?

Reply:
options is frozen, so it would crash